src/components/Header/Header.tsx

```diff
@@ -0,0 +1,90 @@
+import { FC, useState, useEffect } from 'react';
+import { ErrorMessages } from '../../types/ErrorMessages/ErrorMessages';
+import { USER_ID } from '../../api/todos';
+import { Todo } from '../../types/Todo/Todo';
+import { useTodoContext } from '../../utils/hooks/useTodoContext';
+import { isAllTodosComplete } from '../../services/FilterService';
+import { TodoService } from '../../services/TodoService';
+import { getuuidNumber } from '../../utils/uuidNumber';
+import classNames from 'classnames';
+
+export const Header: FC = ({}) => {
+  const { showError, setLoading, todos, inputRef, setTodoTemp, setFocusInput } =
+    useTodoContext();
+  const { createTodo, toggleAllCompleted } = TodoService();
+  const [title, setTitle] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+  };
+
+  const handleSubmit = async (event?: React.FormEvent) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    const normalizeTitle = title.trim();
+
+    if (!normalizeTitle) {
+      showError(ErrorMessages.Empty);
+      setLoading(null);
+      setFocusInput(true);
+
+      return;
+    }
+
+    setTodoTemp(normalizeTitle);
+
+    const todo: Todo = {
+      id: getuuidNumber(),
+      userId: USER_ID,
+      completed: false,
+      title: normalizeTitle,
+    };
+
+    try {
+      setIsSubmitting(true);
+      setLoading([todo.id]);
+      await createTodo(todo);
+      reset();
+    } catch (error) {
+    } finally {
+      setLoading(null);
+      setTodoTemp(null);
+      setIsSubmitting(false);
+    }
+  };
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [isSubmitting, inputRef]);
```

Comment:
you do it a lot of times

But I guess you can create a ref object here instead of Context and use it only here